File: JAVA/Exit Point in a Matrix.java
class Solution {
    private boolean isValid(int[][] matrix, int i, int j) {
        if (i<0 || j <0 || i >= matrix.length || j >= matrix[0].length) return false;
        return true;
    }
    public int[] FindExitPoint(int n, int m, int[][] matrix) {
        int cur = 0;
        int[] ans = new int[2]; 
        int[][] dir = {{0,1}, {1, 0}, {0,-1}, {-1, 0}};
        int i = 0 , j = 0 ;
        while(isValid(matrix, i, j)) {
            if (matrix[i][j] == 1) {
                cur = (cur+1)%4;
                matrix[i][j] = 0;
            }
            i = i + dir[cur][0];
            j = j + dir[cur][1]; 
        }
        return new int[]{i-dir[cur][0], j-dir[cur][1] };
    }
}

File: C++/Exit Point in a Matrix.cpp
class Solution {
  public:
   vector<int> FindExitPoint(int n, int m, vector<vector<int>>& matrix) {
        int i=0, j=0;
        int p_i=0, p_j=0;
        char ch = 'R';
        while(i>=0 and j>=0 and i<n and j<m){
            p_i = i;
            p_j = j;
            if(matrix[i][j]==0){
                if(ch=='R') j++;
                else if(ch=='D') i++;
                else if(ch=='L') j--;
                else i--;
            }
            else{
                matrix[i][j] = 0;
                if(ch=='R'){
                    ch='D';
                    i++;
                }
                else if(ch=='D'){
                    ch='L';
                    j--;
                }
                else if(ch=='L'){
                    ch='U';
                    i--;
                }
                else{
                    ch='R';
                    j++;
                }
            }
        }
        
        return {p_i, p_j};
    }
};
